src/ripple/shamap/impl/SHAMapDelta.cpp

```diff
@@ -112,7 +112,9 @@ bool SHAMap::walkBranch (SHAMapTreeNode* node,
     return true;
 }
 
-bool SHAMap::compare (SHAMap::ref otherMap, Delta& differences, int maxCount)
+bool
+SHAMap::compare (std::shared_ptr<SHAMap> const& otherMap,
+                 Delta& differences, int maxCount)
```

Comment:
@HowardHinnant Agreed. But you should leave a comment in the code. Suggested: `// HINNANT should be const`

@rec Yes you have to audit and make the function thread-safe. This puts you on a collision course with marking mutexes `mutable`.